sdk/logs/src/main/java/io/opentelemetry/sdk/logs/ReadWriteLogRecord.java

```diff
@@ -47,7 +53,87 @@ default ReadWriteLogRecord setAllAttributes(Attributes attributes) {
   /** Return an immutable {@link LogRecordData} instance representing this log record. */
   LogRecordData toLogRecordData();
 
-  // TODO: add additional log record accessors. Currently, all fields can be accessed indirectly via
-  // #toLogRecordData() at the expense of additional allocations.
+  /**
+   * Returns the value of a given attribute if it exists. This is the equivalent of calling
+   * getAttributes().get(key)
+   */
+  @Nullable
+  default <T> T getAttribute(AttributeKey<T> key) {
+    return toLogRecordData().getAttributes().get(key);
+  }
+
+  /** Returns the resource of this log record. */
+  @Override
+  default Resource getResource() {
```

Comment:
This isn't present on `ReadableSpan`. If we add it here, we should minimally open an issue to make sure we add to `ReadableSpan` in a followup.